backend/middleware/accesslog/accesslog_test.go

```diff
@@ -0,0 +1,63 @@
+package log
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"go.uber.org/zap/zaptest"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+
+	middlewarev1 "github.com/lyft/clutch/backend/api/config/middleware/v1"
+	healthcheckv1 "github.com/lyft/clutch/backend/api/healthcheck/v1"
+)
+
+func TestNew(t *testing.T) {
+	tests := []struct {
+		config *middlewarev1.AccessLog
+	}{
+		{config: nil},
+		{config: &middlewarev1.AccessLog{
+			StatusCodeFilter: &middlewarev1.AccessLog_StatusCodeFilter{
+				Operator: middlewarev1.AccessLog_StatusCodeFilter_GE,
+				Value:    uint32(codes.OK),
+			},
+		},
+		},
+	}
+
+	for idx, tt := range tests {
+		tt := tt
+		t.Run(fmt.Sprintf("%d", idx), func(t *testing.T) {
+			t.Parallel()
+
+			m, err := New(tt.config, nil, nil)
+			assert.NoError(t, err)
+			assert.NotNil(t, m)
+		})
+	}
+}
+
+func fakeHandler(ctx context.Context, req interface{}) (interface{}, error) {
+	return &healthcheckv1.HealthcheckResponse{}, nil
+}
+
+func TestInterceptor(t *testing.T) {
```

Comment:
you can use zap's `observer` to record logs and check them during tests

e.g. https://github.com/lyft/clutch/blob/a7a5b625bdaeaceabd2a0f0998bc974eefbae7d4/backend/middleware/timeouts/timeouts_test.go#L120-L121